src/main/java/org/opensearch/ad/rest/handler/ModelValidationActionHandler.java

```diff
@@ -0,0 +1,718 @@
+            long currentInterval = interval.getInterval();
+            if (interval.getUnit() == ChronoUnit.MILLIS) {
+                currentInterval /= 60000;
+            }
+            return currentInterval;
+        }
+
+        @Override
+        public void onResponse(SearchResponse response) {
+            try {
+                Histogram aggregate = checkBucketResultErrors(response);
+                if (aggregate == null) {
+                    return;
+                }
+
+                double fullBucketRate = processBucketAggregationResults(aggregate);
+                long newInterval = (long) Math.ceil(convertIntervalToMinutes(detectorInterval) + 1);
+                // If rate is below success minimum then call search again.
+                if (fullBucketRate > INTERVAL_BUCKET_MINIMUM_SUCCESS_RATE) {
+                    intervalListener.onResponse(this.detectorInterval);
+                } else if (expirationEpochMs < clock.millis()) {
+                    listener
+                        .onFailure(
+                            new AnomalyDetectionException(
+                                "Timed out getting interval recommendation. Please continue with detector creation."
+                            )
+                        );
+                    logger.info("Timed out getting interval recommendation");
+                } else if (newInterval < MAX_INTERVAL_REC_LENGTH_IN_MINUTES) {
+                    this.detectorInterval = new IntervalTimeConfiguration(newInterval, ChronoUnit.MINUTES);
+                    // Searching again using an updated interval
+                    SearchSourceBuilder updatedSearchSourceBuilder = getSearchSourceBuilder(
+                        searchSourceBuilder.query(),
+                        getNewAggregationBuilder(newInterval)
+                    );
+                    client
+                        .search(
+                            new SearchRequest()
+                                .indices(anomalyDetector.getIndices().toArray(new String[0]))
+                                .source(updatedSearchSourceBuilder),
+                            this
+                        );
+                    // this case means all intervals up to max interval recommendation length have been tried
+                    // which further means the next step is to go through A/B validation checks
+                } else {
+                    intervalListener.onResponse(null);
+                }
+
+            } catch (Exception e) {
+                onFailure(e);
+            }
+        }
+
+        @Override
+        public void onFailure(Exception e) {
+            logger.error("Failed to paginate top anomaly results", e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void processIntervalRecommendation(IntervalTimeConfiguration interval, long latestTime) {
+        if (interval == null) {
+            checkRawDataSparsity(latestTime);
+        } else {
```

Comment:
Can you add a clarifying comment about this block? From my understanding this means that no interval could be found with the filters applied (category field, feature fields, etc), and so now you are checking the raw data, to determine if the problem is from the raw data, or the filters